tests/integration/components/code-mirror-test.js

```diff
@@ -0,0 +1,378 @@
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+const SELECTOR_COMPONENT = '[data-test-code-mirror-component]';
+const SELECTOR_EDITOR = `${SELECTOR_COMPONENT} > .cm-editor`;
+const SELECTOR_SCROLLER = `${SELECTOR_EDITOR} > .cm-scroller`;
+const SELECTOR_CONTENT = `${SELECTOR_SCROLLER} > .cm-content`;
+
+module('Integration | Component | code-mirror', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(hbs`<CodeMirror />`);
+    assert.dom(SELECTOR_COMPONENT).exists();
+    assert.dom(SELECTOR_EDITOR).exists();
+    assert.dom(SELECTOR_SCROLLER).exists();
+    assert.dom(SELECTOR_CONTENT).exists();
+    await render(hbs`<CodeMirror>template block text</CodeMirror>`);
+    assert.dom(SELECTOR_COMPONENT).exists();
+    assert.dom(SELECTOR_EDITOR).exists();
+    assert.dom(SELECTOR_SCROLLER).exists();
+    assert.dom(SELECTOR_CONTENT).exists();
+  });
+
+  test('it renders passed document', async function (assert) {
+    const exampleText = 'function myJavaScriptFunction() { return "hello world"; }';
+    this.set('document', exampleText);
+    await render(hbs`<CodeMirror @document={{this.document}} />`);
+    assert.dom(SELECTOR_CONTENT).hasText(exampleText);
+    await render(hbs`<CodeMirror @document={{this.document}}>template block text</CodeMirror>`);
+    assert.dom(SELECTOR_CONTENT).hasText(exampleText);
+  });
+
+  module('Updating passed document', function () {
+    test('it updates rendered document when @document changes', async function (assert) {
+      const exampleText = 'function myJavaScriptFunction() { return "hello world"; }';
+      const exampleTextUpdated = 'export const function myJavaScriptFunction() { return "hello world"; }';
+      this.set('document', exampleText);
+      await render(hbs`<CodeMirror @document={{this.document}} />`);
+      assert.dom(SELECTOR_CONTENT).hasText(exampleText);
+      this.set('document', exampleTextUpdated);
+      assert.dom(SELECTOR_CONTENT).hasText(exampleTextUpdated);
+    });
+    skip('it resets edit history when @presereveHistory is false');
+    skip('it preserves edit history when @presereveHistory is true');
+  });
+
+  module('Updating edited document', function () {
+    skip('it calls @onUpdate when document is edited inside the editor');
+  });
+
+  module('Options', function () {
+    module('allowMultipleSelections', function () {
+      test("it doesn't break the editor when passed", async function (assert) {
+        this.set('allowMultipleSelections', true);
+        await render(hbs`<CodeMirror @allowMultipleSelections={{this.allowMultipleSelections}} />`);
+        assert.dom(SELECTOR_CONTENT).exists();
+        this.set('allowMultipleSelections', false);
+        assert.dom(SELECTOR_CONTENT).exists();
+      });
+      skip('it does something useful with the editor');
+    });
+    module('autocompletion', function () {
```

Comment:
For consistency, I'd place newlines between each `module` invocation, and between each `test`/`skip` block within it. In this file it looks like we're doing that everywhere except within this `Options` module block.

Reply:
Done as well, added line separators!